### DISCUSSION TITLE
Common dependency for both Breeze and Luma

### DISCUSSION THREAD

**Discussion by pk-pankaj** [breezefront/community#247] [3. Integrations]
Hi @vovayatsyuk 

Is there any possibility that we can make a single dependency file/function that can be imported/used in both files Breeze and Luma, and also in the uiComponent file as I have a requirement where I need to write some js for all pages including checkout afterRender so, in this case, I am writing duplicate code?

Thanks,
Pankaj

**Comment by vovayatsyuk** [comment]
No, you don't have to duplicate the code in most cases. Breeze js API is very similar to Luma and allows reusing the same files in many scenarios. Can you share an example of what are you trying to reuse?

**Reply by pk-pankaj** [reply]
Hi,

I have a JS function that checks the form field value and adds class accordingly, so here I want to create a JS file for this function and want to import that in the checkout page JS as well as in breeze extend JS, Currently, I placed that code in both files separately.

This is the JS function code 
```
function checkPrefilledInput(inputParent, inputActive) {
        const fields = document.querySelectorAll(`${inputParent} input, ${inputParent} select, ${inputParent} textarea`);
        fields.forEach(field => {
            if (field.value.length > 0) {
                field.closest(inputParent).classList.add(inputActive);
            }
        });
    }
 
```
 And this is checkout Js file where used duplicate
```
define([
    'uiComponent',
    'ko',
    'jquery',
    'rjsResolver'
], function (Component, ko, $, resolver) {
    'use strict';

    return Component.extend({
        /**
         * @param {HTMLElement} element
         */
        setFormStyle: function (element) {
            var inputParentField = ".field",
                inputActiveClass = "focused",
                _this = this;
            resolver(function () {
                setTimeout(function (){
                    _this.checkPrefilledInput(inputParentField, inputActiveClass);
                }, 500);
            })
        },

        checkPrefilledInput: function (inputParent, inputActive) {
            const fields = document.querySelectorAll(`${inputParent} input, ${inputParent} select, ${inputParent} textarea`);
            fields.forEach(field => {
                if (field.value.length > 0) {
                    field.closest(inputParent).classList.add(inputActive);
                }
            });
        }

    });
}); 
```
Alternatively, is there any other way to check the form fields value to add class to achieve styling globally?

**Reply by vovayatsyuk** [reply]
You can extract your function into separate Breeze and Luma-compatible `Vendor_Module/js/check-prefilled-input.js` file:

```js
define(['jquery'], ($) => {
    function checkPrefilledInput(inputParent, inputActive) {
        ...
    }

    checkPrefilledInput.component = 'Vendor_Module/js/check-prefilled-input';

    return checkPrefilledInput;
})
```

Then use this file in Luma and Breeze.

Source: 
- [Integration example](https://breezefront.com/docs/integration-example#4-prepare-dependencies)
- [Integration tips](https://breezefront.com/docs/integration-tips#migrate-functions)

**Comment by pk-pankaj** [comment]
It helped, thanks @vovayatsyuk
